profiler/src/ProfilerEngine/Datadog.Linux.ApiWrapper/functions_to_wrap.c

```diff
@@ -121,6 +166,90 @@ int dladdr(const void* addr_arg, Dl_info* info)
     return result;
 }
 
+/* Function pointers to hold the value of the glibc functions */
+static int (*__real_execve)(const char* pathname, char* const argv[], char* const envp[]) = NULL;
+
+static char* ddTracePath = NULL;
+
+int execve(const char* pathname, char* const argv[], char* const envp[])
+{
+    if (__real_execve == NULL)
+    {
+        __real_execve = dlsym(RTLD_NEXT, "execve");
+
+        ddTracePath = getenv("DD_TRACE_CRASH_HANDLER");
+
+        if (ddTracePath != NULL && ddTracePath[0] == '\0')
+        {
+            ddTracePath = NULL;
+        }
+    }
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wtautological-compare"
+    if (ddTracePath != NULL && pathname != NULL)
+    {
+        size_t length = strlen(pathname);
+
+        if (length >= 11 && strcmp(pathname + length - 11, "/createdump") == 0)
+        {
+            // Execute the alternative crash handler, and prepend "createdump" to the arguments
+
+            // Count the number of arguments (the list ends with a null pointer)
+            int argc = 0;
+            while (argv[argc++] != NULL);
+
+            // We add two arguments: the path to dd-dotnet, and "createdump"
+            char** newArgv = malloc((argc + 2) * sizeof(char*));
+
+            // By convention, argv[0] contains the name of the executable
+            // Insert createdump as the first actual argument
+            newArgv[0] = ddTracePath;
+            newArgv[1] = "createdump";
+
+            // Copy the remaining arguments
+            memcpy(newArgv + 2, argv, sizeof(char*) * argc);
+
+            size_t envp_count;
+            for (envp_count = 0; envp[envp_count]; ++envp_count);
+            char** new_envp = malloc((envp_count + 1) * sizeof(char*)); // +1 for NULL terminator
+
+            int index = 0;
+
+            for (size_t i = 0; i < envp_count; ++i) {
```

Comment:
Just checking: if the .NET Crash tracking app is crashing while collecting callstacks. We won't be called again because minidump is disabled right ?

Reply:
The goal is to unset `COMPlus_DbgEnableMiniDump` when calling dd-dotnet. The reason is that COMPlus_DbgEnableMiniDump is not supported on NativeAOT, and .NET logs a warning in the console instead of silently ignoring it.

Unsetting CORECLR_ENABLE_PROFILING and LD_PRELOAD is not strictly necessary, but since I wrote the code for COMPlus_DbgEnableMiniDump I added it as a bonus for extra safety.